libavcodec/videotoolbox.c

```diff
@@ -743,11 +743,18 @@ static CFDictionaryRef videotoolbox_decoder_config_create(CMVideoCodecType codec
         if (data)
             CFDictionarySetValue(avc_info, CFSTR("esds"), data);
         break;
-    case kCMVideoCodecType_H264 :
+    case kCMVideoCodecType_H264 : {
+        H264Context *h = avctx->priv_data;
+        if (TARGET_OS_IPHONE && h->ps.sps->frame_mbs_only_flag == 0) {
```

Comment:
I'm assuming tvos is supposed to go down this path as well. Does the check need to be an 'or' with this target or does target_ios_iphone cover both?

Reply:
`TARGET_OS_IPHONE` should cover both (see https://stackoverflow.com/a/49560690)
Also, user from https://github.com/xbmc/xbmc/issues/18287 confirmed that this PR fixed the issue on tvOS.